### PR TITLE
Add a compatibility notice for diesel and the new resolver.

### PR DESCRIPTION
**Pull request by ehuss**
This adds a notice about a compatibility issue with diesel and the new feature resolver. This will display a notice if it detects the scenario where diesel causes a build error.  It also prints a not when using `cargo fix --edition` when it detects a setup that may fail.

### CODE REVIEW

**Comment by ehuss** [comment]
cc @weiznich, does this look reasonable to you?

File: src/cargo/core/compiler/job_queue.rs [hunk 1]
```diff
@@ -1154,4 +1158,60 @@ impl<'cfg> DrainState<'cfg> {
...
+- Add a build-dependency in Cargo.toml on diesel to force Cargo to add the appropriate
+  features. This may look something like this:
+
+    [build-dependencies]
+    diesel = {{ version = \"{}\", features = {} }}
+
+- Try using the previous resolver by setting `resolver = \"1\"` in `Cargo.toml`
+  (see <https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions>
+  for more information).
+- Check for the 2.0 release of diesel which does not have this problem.
```
Comment (weiznich) [inline src/cargo/core/compiler/job_queue.rs:1210]: This warning is only shown if there is an actual error, right? In that case I propose not to have this line here, as there is no 2.0 release yet.

**Review by weiznich — COMMENTED** [review]
This fixes diesels problem, but I wonder if there is some way to write those hints in a more general style so that other crates benefit from that warning as well?

File: src/cargo/core/compiler/job_queue.rs (hunk 1 above)
Comment (ehuss) [reply src/cargo/core/compiler/job_queue.rs]: Sure, I removed it.

**Pull request merged**